### ISSUE TITLE
Declaring in parent, calling in child

### ISSUE THREAD

**Issue by stenio123** [jesusoterogomez/react-notify-toast#2]
Hello, thank you for the effort in this project!

I am trying to use it with react-router 0.13.2 with no success. Could you please provide an example where the <Notification/> tag is declared in the parent, and the "notify" call is called from a child component?

I am trying to have that but running into errors. First I tried adding the <Notification/> tag on my top component, then passed the reference of "notify" as a prop to the children. In this case I get an error - "notify is not defined":

app.js:
`<div className="container">`
          ` <Notifications/>`
          ` <hgroup>`
            `<h1>Welcome!</h1>`
            `<h3>Please login or register.</h3>`
         ` </hgroup>`
          `<div className="LoginRegisterContainer">`
              `<RouteHandler notify={notify}/>`
          `</div>`
`</div>`

and, in the appropriate route:
Login.js
    ...
    login: function (e) {
      e.preventDefault();
      if(canProceed) {
        Auth.login(this.state.email, this.state.password)
          .catch(function(err) {
            var obj = JSON.parse(err.response);
            notify("There's an error logging in: " + obj.message);
            console.log("Error logging in", err);
          });
      } else {
        this.refs.email.isValid();
        this.refs.password.isValid();
      }
    },

I tried using "this.notify", doing a var that = this before the call to Auth.login, and then "that.notify" and still no luck.

Then I removed the <Notification/> tag from the parent app.js, and used it in Login.js, just to test. In this case I get the error 'Uncaught TypeError: (0 , _reactNotifyToast.notify) is not a function'. I checked the React generated code, and it replaces the <Notify> tag with <_reactNotifyToast2>, which might be the problem.... but this _reactNotifyToast2 comes from the fact that I need to import both "Notification" and "notify". 

Could you please advise, or offer an example where the Notification tag is not declared and called in the same component, but declared on a parent and called from a child?

Thanks!

**Comment by jesusoterogomez** [comment]
Hello!, thank you for ( trying :wink: ) using this component. 

I will update this project soon, it needs better documentation, and I have a few more customization and features in mind (that I didn't need at the time, but could be useful for other people).

Speaking of better documentation, The main problem is the distinction between `Notification` and `{notify}`. 

`Notification` is the component that holds the notifications
`{notify}` is the object that contains the function `show()`, it will contain more functions in the future. 

The `Notification` component in your example looks okay. 

I have two use cases, either import `{notify}` where necessary, although in your case you're passing it as a prop in the `RouteHandler`, so it should be accessible from every children that is rendered on this component.

I noticed that in your `Route`, you're not calling notify as a prop, so it should be done in this way.

```jsx
this.props.notify("There's an error logging in: " + obj.message);
```

I also advice that if you're using notify to show an error, you use the second parameter to have it show the error styling.

```js
this.props.notify("There's an error", "error");
```

To call `notify` as a standalone function, you can require `{notify}` and use it, it will still show the notification on the `<Notification/>` component, it doesn't matter if you call `{notify}` on the children component. Like this.

```jsx
import {notify} from 'react-notify-toast'; 
...
someFunction() {
     notify('Sweet!')
}
```

In summary, the notification component is placed on your topmost component and you can use `notify` in any other component in your application as a prop in the router, or as a standalone function.

**Comment by stenio123** [comment]
Thank you for your detailed response! I ended up using a plain old global js library (https://alertifyjs.org/), and called it using window.alertify.error('message'). The problem with this approach is that I wont be able to test it as a React component, but works as a quick fix.

Will give it a try again later, but I believe this.props.notify.show might be the culprit. Thank you again!
